subprojects/frameworklauncher/yarn/src/main/java/com/microsoft/frameworklauncher/common/exit/FrameworkExitSpec.java

```diff
@@ -0,0 +1,334 @@
+// Copyright (c) Microsoft Corporation
+// All rights reserved. 
+//
+// MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated 
+// documentation files (the "Software"), to deal in the Software without restriction, including without limitation 
+// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and 
+// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING 
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND 
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, 
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, 
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE. 
+
+package com.microsoft.frameworklauncher.common.exit;
+
+import com.microsoft.frameworklauncher.common.log.DefaultLogger;
+import com.microsoft.frameworklauncher.common.model.ExitType;
+import com.microsoft.frameworklauncher.common.model.UserContainerExitInfo;
+import com.microsoft.frameworklauncher.common.model.UserContainerExitSpec;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class FrameworkExitSpec {
+  private static final DefaultLogger LOGGER = new DefaultLogger(FrameworkExitSpec.class);
+
+  private static final Map<Integer, FrameworkExitInfo> spec = new TreeMap<>();
+
+  static {
+    // Success ExitCode which is issued from User Container
+    spec.put(FrameworkExitCode.SUCCEEDED.toInt(), new FrameworkExitInfo(
+        FrameworkExitCode.SUCCEEDED.toInt(),
+        null,
+        ExitType.SUCCEEDED,
+        "Succeeded"));
+
+
+    // Failure ExitCode which is issued from Launcher AM to User Container
+    /// Container External Error
+    //// Container Init Failed by YARN
+    spec.put(FrameworkExitCode.CONTAINER_INVALID_EXIT_STATUS.toInt(), new FrameworkExitInfo(
+        -1000,
```

Comment:
Use Enum instead of -1000?

Reply:
Good point, let me try. But some exitcode cannot be got from current hadoop version.